Objective-C/Tests/ReplicatorTest+CustomConflict.m

```diff
@@ -233,6 +233,37 @@ - (void) testConflictResolverMergeDoc {
     AssertEqualObjects(savedDoc.toDictionary, exp);
 }
 
+- (void) testNonBlockingConflictResolver {
+    XCTestExpectation* ex = [self expectationWithDescription: @"testNonBlockingConflictResolver"];
+    NSDictionary* localData = @{@"key1": @"value1"};
+    NSDictionary* remoteData = @{@"key2": @"value2"};
+    [self makeConflictFor: @"doc1" withLocal: localData withRemote: remoteData];
+    [self makeConflictFor: @"doc2" withLocal: localData withRemote: remoteData];
+    
+    TestConflictResolver* resolver;
+    CBLReplicatorConfiguration* pullConfig = [self pullConfig];
+    
+    NSMutableArray* order = [NSMutableArray array];
+    resolver = [[TestConflictResolver alloc] initWithResolver: ^CBLDocument* (CBLConflict* con) {
+        [order addObject: con.localDocument.id];
```

Comment:
NSMutableArray is not thread-safe, so you need to synchronize accesses to`order` inside the resolver block.